### PR TITLE
Add a Danger rule about pushing submodules #trivial

### LINKED ISSUE #2248: OSS - submodule path, client key errors
I'm trying to get setup with the app as OSS.
I'm running into two erros

1. Direct fetching of that commit failed
"parrys-Air:eigen parry$ git submodule update
error: no such remote ref c68592e3fd42eb38fff0b955cc2ea2191e097caa
Fetched in submodule path 'Pods', but it did not contain c68592e3fd42eb38fff0b95"

When trying to run - make oss
screen shot 2017-03-28 at 11 11 26 pm

2.  " CocoaPods-Keys has detected a keys mismatch for your setup.
 What is the key for ArtsyAPIClientKey"

when trying to run - bundle exec pod install

### PR DESCRIPTION
**Pull request by orta**
Fixes #2248 

I added a check to ensure that `danger pr` would work with the project. Removed unused rules, and added another. 

This PR should fail.

### CODE REVIEW

**Comment by orta** [comment]
Aww :(

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          4 Errors
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">No CHANGELOG changes made</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">You need to set the App’s plist version to 3.2.0</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">You need to set the Sticker extensions’s plist version to 3.2.0</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Could not find the commit -e3e9e90ea9c6b8d5c770e14d3831319d028791ea in <a href="https://github.com/artsy/eigen-artefacts/">artsy/eigen-artefacts</a>. &lt;/br&gt;You need to <code>git push</code> in the Pods dir.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_travis">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by orta** [comment]
Alright - this is looking good.

The Dangerfile running on travis uses a Ruby + Linux machine, so we can't use plist buddy. So I switched that work with a `File.read`. This is fine because we don't have to wait for a Mac to be provisioned now. So it's much faster get to responses.

@ashfurrow this removes your new rule, and makes it happen at deploy time, so we don't need the rule.

@alloy I can't _easily_ test that change ^ so it may need debugging on a deploy.

**Comment by orta** [comment]
OK - this is 👍

**Comment by orta** [comment]
Yeah, let's do it

**Comment by orta** [comment]
Nah, let's not, we need some way to kind of say "there pieces need to be in place" iTunes connect + the plist. I think this is a good enough way.

File: Dangerfile [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# This one runs on Travis CI
+# There is another in Dangerfuile.circle.rb
```
Comment (Haroenv) [inline Dangerfile]: Typo: should be `Dangerfile.circle.rb`

Comment (orta) [inline Dangerfile]: Fixed! Thanks 👯‍♂️

**Comment by alloy** [comment]
Excellent

**Pull request merged**